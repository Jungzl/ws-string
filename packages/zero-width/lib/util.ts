/**
 * Created by user on 2019/8/12.
 */

/**
 * simple merge, not real regexp merge
 * @private
 */
export function _regexpMerge(re: (string | RegExp)[])
{
	return new RegExp('[' + re.map(r => (typeof r === 'string' ? r : r.source).replace(/^\[/, '').replace(/\]$/, ''))
		.join('') + ']', 'ug')
}
