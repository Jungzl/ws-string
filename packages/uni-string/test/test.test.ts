/**
 * Created by user on 2018/11/8/008.
 */

import UString from '../src';
import { inspect } from 'util';

it(`UString = UString.UString = UString.default`, done =>
{

	expect(UString).toEqual(UString.UString);
	expect(UString).toEqual(UString.default);

	done();
});

describe(`simple`, () =>
{

	([
		[`𠬠典`, ['𠬠', '典']],
		[`𡬶寻尋`, ['𡬶', '寻', '尋']],
	] as [string, string[]][]).forEach(function ([input, expected])
	{

		it(inspect(input), () =>
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			let actual = UString.split(input, '');

			let size = UString.size(input);

			expect(actual).toEqual(expected);
			expect(actual.length).toEqual(expected.length);
			expect(size).toEqual(expected.length);

			let arr = UString.slice(input, 1, 2);

			expect(arr).toEqual(expected[1]);

			const _arr_input = input.split('');
			console.dir(_arr_input);
			expect(actual).toMatchSnapshot();

			expect(input.length).toMatchSnapshot();
			expect(size).toMatchSnapshot();

			expect(arr).toMatchSnapshot();

		});
	})

});

describe(`demo`, () =>
{

	test('mixin', () =>
	{

		let t = (new UString('♥️𠬠典')).padStart(10, '𠬠');

		t = t.concat('\n♥️𠬠典そこで彼らは\'"\'👩‍👩‍👧‍👦\'，オリーブ山と呼ばれる山からエルサレムに帰った。');

		expect(t).toMatchSnapshot();
		expect(t.split('')).toMatchSnapshot();

		let t2 = new UString(t);

		expect(t2).toMatchSnapshot();
		expect(t2.length).toMatchSnapshot();
		expect(t2.charLength).toMatchSnapshot();
		expect(t2.size()).toMatchSnapshot();

		expect(t2.indexOf('𠬠典', 2)).toMatchSnapshot();
		expect(t2.endsWith('𠬠典')).toMatchSnapshot();

		expect(t2.split('')).toMatchSnapshot();

	});

	test(inspect('👩‍👩‍👧‍👦'), () =>
	{

		let t3 = new UString('👩‍👩‍👧‍👦');

		expect(t3).toMatchSnapshot();
		expect(t3.length).toMatchSnapshot();
		expect(t3.charLength).toMatchSnapshot();
		expect(t3.size()).toMatchSnapshot();

		expect(t3.split('')).toMatchSnapshot();

	});

})

test('support', () => {

	let actual = UString.support;

	let keys = Object.keys(actual);

	console.dir(actual)

	expect(keys).toMatchSnapshot();

});
