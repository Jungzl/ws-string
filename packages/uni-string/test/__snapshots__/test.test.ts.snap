// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo '­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд' 1`] = `
UString {
  "0": "№┐й",
  "1": "№┐й",
  "10": "№┐й",
  "2": "РђЇ",
  "3": "№┐й",
  "4": "№┐й",
  "5": "РђЇ",
  "6": "№┐й",
  "7": "№┐й",
  "8": "РђЇ",
  "9": "№┐й",
}
`;

exports[`demo '­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд' 2`] = `11`;

exports[`demo '­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд' 3`] = `1`;

exports[`demo '­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд' 4`] = `1`;

exports[`demo '­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд' 5`] = `
Array [
  "­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд",
]
`;

exports[`demo mixin 1`] = `
"­ага­ага­ага­ага­ага­ага­агаРЎЦ№ИЈ­агатЁИ
РЎЦ№ИЈ­агатЁИсЂЮсЂЊсЂДтй╝сѓЅсЂ»'\\"'­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд'№╝їсѓфсЃфсЃ╝сЃќт▒▒сЂетЉ╝сЂ░сѓїсѓІт▒▒сЂІсѓЅсѓесЃФсѓхсЃгсЃасЂФтИ░сЂБсЂЪсђѓ"
`;

exports[`demo mixin 2`] = `
Array [
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "№┐й",
  "РЎЦ",
  "№ИЈ",
  "№┐й",
  "№┐й",
  "тЁИ",
  "
",
  "РЎЦ",
  "№ИЈ",
  "№┐й",
  "№┐й",
  "тЁИ",
  "сЂЮ",
  "сЂЊ",
  "сЂД",
  "тй╝",
  "сѓЅ",
  "сЂ»",
  "'",
  "\\"",
  "'",
  "№┐й",
  "№┐й",
  "РђЇ",
  "№┐й",
  "№┐й",
  "РђЇ",
  "№┐й",
  "№┐й",
  "РђЇ",
  "№┐й",
  "№┐й",
  "'",
  "№╝ї",
  "сѓф",
  "сЃф",
  "сЃ╝",
  "сЃќ",
  "т▒▒",
  "сЂе",
  "тЉ╝",
  "сЂ░",
  "сѓї",
  "сѓІ",
  "т▒▒",
  "сЂІ",
  "сѓЅ",
  "сѓе",
  "сЃФ",
  "сѓх",
  "сЃг",
  "сЃа",
  "сЂФ",
  "тИ░",
  "сЂБ",
  "сЂЪ",
  "сђѓ",
]
`;

exports[`demo mixin 3`] = `
UString {
  "0": "№┐й",
  "1": "№┐й",
  "10": "№┐й",
  "11": "№┐й",
  "12": "№┐й",
  "13": "№┐й",
  "14": "РЎЦ",
  "15": "№ИЈ",
  "16": "№┐й",
  "17": "№┐й",
  "18": "тЁИ",
  "19": "
",
  "2": "№┐й",
  "20": "РЎЦ",
  "21": "№ИЈ",
  "22": "№┐й",
  "23": "№┐й",
  "24": "тЁИ",
  "25": "сЂЮ",
  "26": "сЂЊ",
  "27": "сЂД",
  "28": "тй╝",
  "29": "сѓЅ",
  "3": "№┐й",
  "30": "сЂ»",
  "31": "'",
  "32": "\\"",
  "33": "'",
  "34": "№┐й",
  "35": "№┐й",
  "36": "РђЇ",
  "37": "№┐й",
  "38": "№┐й",
  "39": "РђЇ",
  "4": "№┐й",
  "40": "№┐й",
  "41": "№┐й",
  "42": "РђЇ",
  "43": "№┐й",
  "44": "№┐й",
  "45": "'",
  "46": "№╝ї",
  "47": "сѓф",
  "48": "сЃф",
  "49": "сЃ╝",
  "5": "№┐й",
  "50": "сЃќ",
  "51": "т▒▒",
  "52": "сЂе",
  "53": "тЉ╝",
  "54": "сЂ░",
  "55": "сѓї",
  "56": "сѓІ",
  "57": "т▒▒",
  "58": "сЂІ",
  "59": "сѓЅ",
  "6": "№┐й",
  "60": "сѓе",
  "61": "сЃФ",
  "62": "сѓх",
  "63": "сЃг",
  "64": "сЃа",
  "65": "сЂФ",
  "66": "тИ░",
  "67": "сЂБ",
  "68": "сЂЪ",
  "69": "сђѓ",
  "7": "№┐й",
  "8": "№┐й",
  "9": "№┐й",
}
`;

exports[`demo mixin 4`] = `70`;

exports[`demo mixin 5`] = `49`;

exports[`demo mixin 6`] = `49`;

exports[`demo mixin 7`] = `8`;

exports[`demo mixin 8`] = `false`;

exports[`demo mixin 9`] = `
Array [
  "­ага",
  "­ага",
  "­ага",
  "­ага",
  "­ага",
  "­ага",
  "­ага",
  "РЎЦ№ИЈ",
  "­ага",
  "тЁИ",
  "
",
  "РЎЦ№ИЈ",
  "­ага",
  "тЁИ",
  "сЂЮ",
  "сЂЊ",
  "сЂД",
  "тй╝",
  "сѓЅ",
  "сЂ»",
  "'",
  "\\"",
  "'",
  "­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд",
  "'",
  "№╝ї",
  "сѓф",
  "сЃф",
  "сЃ╝",
  "сЃќ",
  "т▒▒",
  "сЂе",
  "тЉ╝",
  "сЂ░",
  "сѓї",
  "сѓІ",
  "т▒▒",
  "сЂІ",
  "сѓЅ",
  "сѓе",
  "сЃФ",
  "сѓх",
  "сЃг",
  "сЃа",
  "сЂФ",
  "тИ░",
  "сЂБ",
  "сЂЪ",
  "сђѓ",
]
`;

exports[`demo ­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд 1`] = `
UString {
  "0": "№┐й",
  "1": "№┐й",
  "10": "№┐й",
  "2": "РђЇ",
  "3": "№┐й",
  "4": "№┐й",
  "5": "РђЇ",
  "6": "№┐й",
  "7": "№┐й",
  "8": "РђЇ",
  "9": "№┐й",
}
`;

exports[`demo ­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд 2`] = `11`;

exports[`demo ­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд 3`] = `1`;

exports[`demo ­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд 4`] = `1`;

exports[`demo ­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд 5`] = `
Array [
  "­ЪЉЕРђЇ­ЪЉЕРђЇ­ЪЉДРђЇ­ЪЉд",
]
`;

exports[`simple '­агатЁИ' 1`] = `
Array [
  "№┐й",
  "№┐й",
  "тЁИ",
]
`;

exports[`simple '­агатЁИ' 2`] = `
Array [
  "­ага",
  "тЁИ",
]
`;

exports[`simple '­агатЁИ' 3`] = `3`;

exports[`simple '­агатЁИ' 4`] = `2`;

exports[`simple '­агатЁИ' 5`] = `"тЁИ"`;

exports[`simple '­АгХт»╗т░І' 1`] = `
Array [
  "№┐й",
  "№┐й",
  "т»╗",
  "т░І",
]
`;

exports[`simple '­АгХт»╗т░І' 2`] = `
Array [
  "­АгХ",
  "т»╗",
  "т░І",
]
`;

exports[`simple '­АгХт»╗т░І' 3`] = `4`;

exports[`simple '­АгХт»╗т░І' 4`] = `3`;

exports[`simple '­АгХт»╗т░І' 5`] = `"т»╗"`;

exports[`support 1`] = `
Array [
  "constructor",
  "split",
  "substr",
  "substring",
  "indexOf",
  "includes",
  "slice",
  "charAt",
  "startsWith",
  "endsWith",
  "padEnd",
  "padStart",
  "codePointAt",
]
`;
